Frontend/src/components/visualization/high-level-graph-visualization.tsx

```diff
@@ -20,310 +20,239 @@ import { IGraphDetails } from "@/types/graph-types";
+                                        selectedNode,
+                                        setSelectedNode
+                                    }: Props) => {
+    const [loading, setLoading] = useState(false);
+    const networkContainerRef = useRef<HTMLDivElement>(null);
+    const [network, setNetwork] = useState<Network | null>(null);
+    const nodesRef = useRef<any>(new DataSet([]));
+    const edgesRef = useRef<any>(new DataSet([]));
+    const [vertexCountExceed, setVertexCountExceed] = useState<boolean>(false);
+
+    // Helper: scale vertexcount to node size
+    const getNodeSize = (vertexCount: number, maxVertexCount: number) => {
+        const minSize = 20;
+        const maxSize = 100;
+        const size = minSize + ((vertexCount / maxVertexCount) * (maxSize - minSize));
+        return size;
+    };
 
-  const loadHighLevelEdgeList = (): IEdge[] => {
-    const edges: IEdge[] = [];
-    const partitionCount = graph?.partitions.length || 0;
-    for (let i = 0; i < partitionCount; i++) {
-      for (let j = i + 1; j <= partitionCount; j++) {
-        edges.push({ from: i, to: j, properties: `${i}-${j} edge` });
-      }
-    }
+    const loadHighLevelEdgeList = (): IEdge[] => {
+        const edges: IEdge[] = [];
+        const partitionCount = graph?.partitions.length || 0;
+        for (let i = 0; i < partitionCount; i++) {
+            for (let j = i + 1; j < partitionCount; j++) {
+                edges.push({ from: graph!.partitions[i].idpartition, to: graph!.partitions[j].idpartition });
+            }
+        }
+        return edges;
+    };
 
-    return edges;
-  };
+    const loadHighLevelView = async () => {
+        try {
+            setLoading(true);
+            nodesRef.current.clear();
+            edgesRef.current.clear();
+
+            if (graph && graph.partitions && Array.isArray(graph.partitions)) {
+                // Create nodes with size proportional to vertexcount
+                const maxVertexCount = Math.max(...graph.partitions.map(p => p.vertexcount));
+
+                const validatedNodes = graph.partitions.map((partition) => ({
+                    id: partition.idpartition,
+                    label: partition.idpartition.toString(),
+                    shape: "dot",
+                    color: "#97c2fc",
+                    size: getNodeSize(partition.vertexcount, maxVertexCount),
+                }));
+                // Create edges
+                const validatedEdges = loadHighLevelEdgeList();
+
+                edgesRef.current.add(validatedEdges);
+                nodesRef.current.add(validatedNodes);
+
+                if (network) {
+                    network.setData({ nodes: nodesRef.current, edges: edgesRef.current });
+                    network.fit();
+                }
+            } else {
+                // console.warn(ERROR_MSG.noData, selectedNode);
+                // message.warning(ERROR_MSG.noData);
```

Comment:
Shall we remove the complete else block, because it has only two commented lines only?